File: src/Synergy.Recruitment.Api/ClientApp/src/app/@shared/factories/candidate-factory.ts
import { CandidateStatusPresent } from '../present-models/candidates/candidate-status.present';
import { CandidateStatus } from '../data-models/candidates/candidate-status';

/**
 * Contains static mapping methods to map @see Candidate related instances.
 * @export
 * @class CandidateFactory
 */
export  class CandidateFactory {

    /**
     * Creates @see CandidateStatusPresent model from @see CandidateStatus.
     * @static
     * @param {CandidateStatus} candidate
     * @returns {CandidateStatusPresent}
     * @memberof CandidateFactory
     */
    public static getPresent(candidate: CandidateStatus): CandidateStatusPresent {
      return {
        id: candidate.id,
        firstName: candidate.firstName,
        lastName: candidate.lastName,
        position: candidate.position,
        updatedBefore: '',
      } as CandidateStatusPresent;
    }
  }

File: src/Synergy.Recruitment.Business/Factories/CandidateFactory.cs
using System;
using System.Linq;

using Synergy.Recruitment.Data.Models;
using Synergy.Recruitment.Rest.Models.Candidate;

namespace Synergy.Recruitment.Business.Factories
{
    /// <summary>
    /// The factory class containing static methods for instantiating <see cref="Candidate"/> related instances.
    /// </summary>
    public class CandidateFactory
    {
        /// <summary>
        /// Gets the <see cref="CandidateResponse"/> out of <see cref="Candidate"/>.
        /// </summary>
        public static Func<Candidate, CandidateResponse> GetCandidateResponse
            => technology
                => new CandidateResponse
                {
                    Id = technology.Id,
                    // FirstName = technology.FirstName,
                    // LastName = technology.LastName,
                    // Position = MockFactory.GetPositions().FirstOrDefault(p => p.Id == technology.PositionId).Name,
                    UpdatedOn = technology.UpdatedOn,
                };
    }
}
